wl/server/MuninStatistics.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2021 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
+package wl.server;
+
+import java.io.*;
+import java.util.*;
+import wl.utils.*;
+
+public class MuninStatistics {
+	public static final MuninStatistics MUNIN = new MuninStatistics();
+
+	public static void handleMunin(InputStream in, PrintStream out) throws Exception {
+		final int version = Integer.valueOf(ServerUtils.readLine(in));
+		if (version != 1)
+			throw new ServerUtils.WLProtocolException("Unsupported munin version '" + version +
+			                                          "' (only supported version is '1')");
+		ServerUtils.log("Munin version: " + version);
+		ServerUtils.checkEndOfStream(in);
+
+		ServerUtils.passwordAuthentification(in, out, Utils.config("muninpassword"));
```

Comment:
The only reason Widelands uses a hash instead of the plaintext password is that your credentials are stored in your config file where having the plaintext password is suboptimal (and IIRC in the past the password hashing was the *only* encryption). But the actual safety is provided by the CHAP2 protocol (checksumming a "magic string" plus a unique random number) so it does not matter if this magic string is a password, a hash, a hash's hash, or whatever. 

And yes this login will mainly be done from localhost.